components/products/VariantSelector.tsx

```
@@ -0,0 +1,160 @@
+  return (
+    <div className="space-y-4">
+      {Object.entries(variantsByType).map(([type, typeVariants]) => (
+        <div key={type} className="space-y-2">
+          <label className="text-sm font-medium capitalize">
+            {type}
+            {typeVariants.some((v) => !v.inStock) && (
+              <span className="ml-2 text-xs text-muted-foreground">
+                (Some options out of stock)
+              </span>
+            )}
+          </label>
```

Biome (2.3.14)

[error] A form label must be associated with an input.

Consider adding a `for` or `htmlFor` attribute to the label element or moving the input element to inside the label element.

(lint/a11y/noLabelWithoutControl)